### ISSUE TITLE
`unused_async_trait_impl` also fires on inherent methods, duplicating `unused_async`

### ISSUE THREAD

**Issue by dennisameling** [rust-lang/rust-clippy#17772]
### Summary

`unused_async_trait_impl` is documented as checking *trait* method implementations, and `unused_async` skips trait methods so the two don't overlap. But `unused_async_trait_impl` also fires on `async fn`s in inherent `impl` blocks, so the same function gets both warnings. `check_impl_item` doesn't check that the parent `impl` is a trait impl: https://github.com/rust-lang/rust-clippy/blob/f0c668aa82a404de25d4f331210e018b3eb8feed/clippy_lints/src/unused_async.rs#L265

### Lint Name

unused_async_trait_impl

### Reproducer

I tried this code:

```rust
#![warn(clippy::unused_async, clippy::unused_async_trait_impl)]

pub struct S;

impl S {
    pub async fn get(&self) -> u32 {
        1
    }
}
```

I saw this happen:

```
warning: unused `async` for async trait impl function with no `.await` statements
warning: unused `async` for function with no await statements
```

I expected to see this happen: only the `unused_async` warning, since `S::get` is not a trait impl.

### Version

```text
rustc 1.98.1 (48a229cea 2026-09-01)
binary: rustc
commit-hash: 48a229ceaefd4985c50990b14116b6d856af0985
commit-date: 2026-09-01
host: x86_64-unknown-linux-gnu
release: 1.98.1
LLVM version: 22.1.8

Also reproduces with `nightly-2026-09-19` (clippy 0.1.100).
```
